arch/arm/core/cortex_m/mpu/Kconfig

```diff
@@ -38,3 +38,10 @@ config NXP_MPU
 	default n
 	help
 	  MCU has NXP MPU
+
+config MPU_ALLOW_FLASH_WRITE
+	bool "Add MPU access to write to flash"
+	depends on NXP_MPU
+	default n
```

Comment:
This sounds fine, my only complaint is the naming.  I don't like options where you have DISABLE and say Y.  It's negative logic.  I'd rather have a ENABLE=y, so FLASH_ENABLE_WRITE_SUPPORT.